### ISSUE TITLE
The OpenShift client should behave like a regular Kubernetes client when connected to a plain k8s cluster

### ISSUE THREAD

**Issue by metacosm** [fabric8io/kubernetes-client#2856]
There might be other instances but at least `getVersion` doesn't work properly: it returns `null` when connected to a plain k8s cluster when it should return the cluster version.

**Comment by rohanKanojia** [comment]
I think the OpenShift version, and Kubernetes versions are two different things and they're being requested at two different URLs. If we want to use OpenShiftClient on top of Kubernetes, why can't we adapt to KubernetesClient and use it for that particular scenario?

**Comment by metacosm** [comment]
The `getVersion` method is defined in the `KubernetesClient` interface so I'm not sure what you mean by "adapt to KubernetesClient". The fact that the OpenShift implementation hits a different endpoint is exactly the issue: calling the `getVersion` method which is part of the `KubernetesClient` contract should return the Kubernetes version even when the client is really an OpenShift client.

**Comment by rohanKanojia** [comment]
ohk, I agree we should not return `null`.

**Comment by metacosm** [comment]
> ohk, I agree we should not return `null`. But I'm not sure if returning a Kubernetes version is a good idea or not. You're using OpenShiftClient and you asked for OpenShift version. I think we should throw an exception that OpenShift cluster could not be detected.

I disagree. You're calling a method on the `KubernetesClient` interface. That method should return something valid for all implementations. If this method is expected to fail for the `OpenShiftClient` in case it's connected to a plain k8s cluster, then this method shouldn't be on the `KubernetesClient` interface and be called `getOpenShiftVersion` or something similar.
